public/components/health-check/health-check.tsx

```diff
@@ -1,367 +1,142 @@
-                },
-                {
-                    id: 2,
-                    title: 'Check Elasticsearch index pattern',
-                    description: checks.pattern ? <span><EuiLoadingSpinner size="m" /> Checking...</span> : 'Disabled'
-                },
-                {
-                    id: 3,
-                    title: 'Check Elasticsearch template',
-                    description: checks.template ? <span><EuiLoadingSpinner size="m" /> Checking...</span> : 'Disabled'
-                },
-                {
-                    id: 4,
-                    title: 'Check index pattern fields',
-                    description: checks.fields ? <span><EuiLoadingSpinner size="m" /> Checking...</span> : 'Disabled'
-                }
-            );
-            this.setState({ checks, results },
-                async () => {
-                    await Promise.all([this.checkPatterns(), this.checkApiConnection()]);
-                    if (checks.fields) {
-                        const i = results.map(item => item.id).indexOf(4);
-                        try {
-                            await PatternHandler.refreshIndexPattern();
-                            results[i].description = <span><EuiIcon type="check" color="secondary" ></EuiIcon> Ready</span>;
-                            this.setState({ results });
-                        } catch (error) {
-                            results[i].description = <span><EuiIcon type="alert" color="danger" ></EuiIcon> Error</span>;
-                            this.setState({ results }, () => this.handleError(error));
-                        }
-                    }
-
-                    if (!this.state.errors || !this.state.errors.length) {
-                        setTimeout(() => {
-                            const params = this.$rootScope.previousParams || {};
-                            var queryString = Object.keys(params).map(key => key + '=' + params[key]).join('&');
-                            const url = 'wazuh#' + (this.$rootScope.previousLocation || '') + '?' + queryString;
-                            window.location.assign(
-                                chrome.addBasePath(url)
-                            );
-                            return;
-                        }, 300);
-                    }
-                    return;
-                });
-        } catch (error) {
-            this.handleError(error);
-        }
-    }
+  const handleError = (errors, parsed) => {
+    let _errors = checkErrors;
+    _errors.push(
+      ...(parsed
+        ? errors.map((error) => ErrorHandler.handle(error, 'Health Check', { silent: true }))
+        : errors)
+    );
+    setCheckErrors(_errors);
+  };
 
-    goApp() {
-        window.location.href = '/app/wazuh#/settings';
-    }
+  const showToast = (color, title, text, time) => {
+    toasts.add({
+      color: color,
+      title: title,
+      text: text,
+      toastLifeTimeMs: time,
+    });
+  };
```

Comment:
I have seen this type of function, repeated in different files. Maybe, we should keep the same API for the toasts services, and remove the custom function to trigger a toast.